computer_vision/intensity_based_segmentation.py

```diff
@@ -0,0 +1,62 @@
+# Source: "https://www.ijcse.com/docs/IJCSE11-02-03-117.pdf"
+
+# Importing necessary libraries
+import numpy as np
+from PIL import Image
+import matplotlib.pyplot as plt
+
+
+def segment_image(image: np.ndarray, thresholds: list[int]) -> np.ndarray:
+    """
+    Performs image segmentation based on intensity thresholds.
+
+    Args:
+        image (np.ndarray): Input grayscale image as a 2D array.
+        thresholds (list[int]): Intensity thresholds to define segments.
+
+    Returns:
+        np.ndarray: A labeled 2D array where each region corresponds to a threshold range.

```

Comment:
Please do not repeat the datatypes that are in the function signature and verified by `mypy`. Duplicates are problematic when one is changed and the other is not, and the reader does not know which one to trust.
```suggestion
        image: Input grayscale image as a 2D array.
        thresholds: Intensity thresholds to define segments.

    Returns:
        A labeled 2D array where each region corresponds to a threshold range.
```